### PR TITLE
Add sendPush function using web-push for payload encryption

### PR DESCRIPTION
**Pull request by JayChl**
- Need to generate vapid for first execution #286
- Need to get server vapid public key #287 
- Need to make testcase #285

### CODE REVIEW

File: server/push/controllers/send_push.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import webpush from 'web-push';
+
+/**
+ * Send push with payload to endpoint with it's authentication key.
+ * @param {string} endpoint endpoint from subscription
+ * @param {string} keys authentication key and p256dh key from subscription
+ * @param {string} payload payload data send to endpoint
+ * @return {promise} result includes statusCode, headers, body
+ */
+function sendPush(endpoint, keys, payload) {
```
Comment (romandev) [inline server/push/controllers/send_push.js]: I think you can use |export| in ECMA6 feature as follows:
```
export function sendPush(endpoint, keys, payload) {
    ...
}
```

File: server/push/controllers/send_push.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    webpush.sendNotification(pushSubscription, payload)
+    .then((webPushResult) => {
+      return resolve(webPushResult);
+    }).catch((webPushError) => {
+      return reject(webPushError);
+    });
+});
+}
+
+module.exports = sendPush;
```
Comment (romandev) [inline server/push/controllers/send_push.js]: If you use export function, then this is unnecessary.

File: server/push/controllers/send_push.js (hunk 1 above)
Comment (JayChl) [inline server/push/controllers/send_push.js]: Done! Thanks for the information

File: server/push/controllers/send_push.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ * @param {string} endpoint endpoint from subscription
+ * @param {string} keys authentication key and p256dh key from subscription
+ * @param {string} payload payload data send to endpoint
+ * @return {promise} result includes statusCode, headers, body
+ */
+function sendPush(endpoint, keys, payload) {
+  return new Promise((resolve, reject) => {
+    webpush.setGCMAPIKey(config.serverInfo.pushServerKey);
+    webpush.setVapidDetails(
+      'https://github.com/romandev/absolute',
```
Comment (JayChl) [inline server/push/controllers/send_push.js]: I tried, but web-push throws error for here
  if (typeof subject !== 'string' || subject.length === 0) {
    throw new Error('The subject value must be a string containing a URL or ' +
      'mailto: address. ' + subject);
  }
So, use variable for the parameter to readability

File: server/push/controllers/send_push.js (hunk 2 above)
Comment (JayChl) [inline server/push/controllers/send_push.js]: Done

File: server/push/controllers/send_push.js (hunk 3 above)
Comment (romandev) [inline server/push/controllers/send_push.js]: Thank you for detailed explanation!

File: server/push/controllers/send_push.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2017 The Absolute Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import config from '../../config';
+import webpush from 'web-push';
+
+const serverSubject = 'https://github.com/romandev/absolute';
```
Comment (romandev) [inline server/push/controllers/send_push.js]: Let's move this subject into function and I think this should be uppercase only?
const SERVER_SUBJECT = ...
like this?

**Pull request merged**